packages/client/src/shared/libs/tests/socket-test.ts

```diff
@@ -0,0 +1,126 @@
+import { io } from 'socket.io-client';
+
+export default class Tester {
+  socketObject: any;
+
+  constructor(socketObject: any) {
+    this.socketObject = socketObject;
+  }
+
+  async connectTimeTest(type: 'websocket' | 'socket-io', serverUrl: string) {
+    if (type === 'websocket') {
+      const startTime = performance.now();
+      const ws = new WebSocket(serverUrl);
+      ws.onopen = () => {
+        const endTime = performance.now();
+        const connectionTime = endTime - startTime;
+        console.log('WebSocket Connection Time: ', connectionTime, 'ms');
+      };
+    } else {
+      const startTime = performance.now();
+      const socket = io(serverUrl, {
+        transports: ['websocket'],
+      });
+      socket.on('connect', () => {
+        const endTime = performance.now();
+        const connectionTime = endTime - startTime;
+        console.log('Socket.io Connection Time: ', connectionTime, 'ms');
+      });
+    }
+  }
+
+  async latencyTest(
+    type: 'websocket' | 'socket-io',
+    messageCount: number = 100,
+    messageType: string,
+    args?: any,
+  ) {
+    const results = {
+      average: 0,
+      min: Infinity,
+      max: 0,
+      samples: [] as number[],
+    };
+
+    const pingPong = async (): Promise<number> => {
+      return new Promise((resolve) => {
+        const startTime = performance.now();
+        if (type === 'socket-io') {
+          this.socketObject.emit(messageType, args);
+          this.socketObject.on(messageType, () => {
+            const latency = performance.now() - startTime;
+            resolve(latency);
+          });
+        } else {
+          this.socketObject.send(messageType, args);
+          this.socketObject.onmessage = () => {
+            const latency = performance.now() - startTime;
+            resolve(latency);
+          };
+        }
+      });
+    };
+
+    for (let i = 0; i < messageCount; i++) {
+      const latency = await pingPong();
+      results.samples.push(latency);
+      results.average += latency / messageCount;
+      results.min = Math.min(results.min, latency);
+      results.max = Math.max(results.max, latency);
+    }
```

Comment:
for문 말고 고차함수로 표현하는 게 훨씬 보기 편할 거 같아요!
```suggestion
    Array.from({length: messageCount}).forEach()
```